### PR TITLE
SharedCounter uses ISharedObjectKind

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

SharedCounter now uses ISharedObjectKind and does not export the class

## Breaking Changes

Most users of `SharedCounter` should be unaffected as long as the stick to the factory patterns supported by ISharedObjectKind.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md [hunk 1]
```diff
@@ -4,8 +4,6 @@
 
 ```ts
 
-/// <reference types="mocha" />
```
Comment (Abe27342) [inline packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md:7]: IIRC this is some incremental API-extractor bug?

File: packages/service-clients/azure-client/api-report/azure-client.api.md [hunk 2]
```diff
@@ -68,7 +68,7 @@ export interface AzureContainerVersion {
 
 // @internal @deprecated
 export class AzureFunctionTokenProvider implements ITokenProvider {
-    constructor(azFunctionUrl: string, user?: Pick<AzureMember<any>, "additionalDetails" | "userId" | "userName"> | undefined);
+    constructor(azFunctionUrl: string, user?: Pick<AzureMember<any>, "userId" | "userName" | "additionalDetails"> | undefined);
```
Comment (Abe27342) [inline packages/service-clients/azure-client/api-report/azure-client.api.md:71]: might be a similar case to mocha-test-setup? Haven't seen this one specifically before (though I vaguely recall a similar api-extractor issue around random reordering along these lines... if CI passes, not a big deal)

File: .changeset/tidy-rules-fall.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"@fluidframework/counter": minor
+---
+
+SharedCounter now uses ISharedObjectKind and does not export the class
+
+Most users of `SharedCounter` should be unaffected as long as the stick to the factory patterns supported by ISharedObjectKind.
```
Comment (Abe27342) [inline .changeset/tidy-rules-fall.md:7]:
```suggestion
Most users of `SharedCounter` should be unaffected as long as they stick to the factory patterns supported by ISharedObjectKind.
```

**Review by Abe27342 — APPROVED** [review]

File: packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md (hunk 1 above)
Comment (CraigMacomber) [reply packages/test/mocha-test-setup/api-report/mocha-test-setup.api.md:7]: Ya, I'll rever

File: packages/service-clients/azure-client/api-report/azure-client.api.md (hunk 2 above)
Comment (CraigMacomber) [reply packages/service-clients/azure-client/api-report/azure-client.api.md]: reverted

**Review by jikim-msft — APPROVED** [review]

**Pull request merged**